python/mxnet/random.py

```diff
@@ -38,7 +38,13 @@ def uniform(low, high, shape=None, ctx=None, out=None):
         if isinstance(shape, int):
             shape = (shape,)
         out = empty(shape, ctx)
-    return NDArray._sample_uniform(low=low, high=high, shape=out.shape, out=out)
+    if out.dtype == np.float32 or out.dtype ==np.float64:
+        return NDArray._sample_uniform(low=low, high=high, shape=out.shape, out=out)
+    else:
+        tmp = empty(out.shape, out.ctx)
+        tmp = NDArray._sample_uniform(low=low, high=high, shape=out.shape, out=tmp)
+        out[:] = tmp
+        return out
```

Comment:
if u are trying to compute use int or some other types like int, long double. that depends on underlining shadow. And for GPU side it could be big issue as cudnn is not able to do so.